spp_change_request/models/change_request.py

```diff
@@ -140,6 +148,42 @@ class ChangeRequestBase(models.Model):
         auto_join=True,
     )
 
+    @api.onchange("request_type")
+    def _onchange_request_type(self):
+        self._compute_request_type_target()
+        self.registrant_id = None
+
+    def _compute_request_type_target(self):
+        request_type_target = None
+        for rec in self:
+            if rec.request_type:
+                request_type_targets = self.env["spp.change.request.targets"].search(
+                    [("name", "=", rec.request_type)], limit=1
+                )
+                if request_type_targets:
+                    request_type_target = request_type_targets.id
+
+            rec.request_type_target = request_type_target
```

Comment:
@emjay0921 , please move the `request_type_target = None` inside the for loop to reinitialize the variable every loop. Adding it outside then using it to a record's field may result to data error.

```suggestion
    def _compute_request_type_target(self):
        for rec in self:
            request_type_target = None
            if rec.request_type:
                request_type_targets = self.env["spp.change.request.targets"].search(
                    [("name", "=", rec.request_type)], limit=1
                )
                if request_type_targets:
                    request_type_target = request_type_targets.id

            rec.request_type_target = request_type_target
```